client/src/user-context.tsx

```diff
@@ -193,45 +214,19 @@ export function UserDataProvider(props: { children: React.ReactNode }) {
         },
         maintenance: data.maintenance,
         settings,
-        offlineSettings: null,
+        offlineSettings: OfflineSettingsData.read(),
         mutate,
       };
+      setSessionStorageData(user);
+      handleOffline(user);
+      return user;
     }
   );
 
   React.useEffect(() => {
     removeDeprecatedLocalStorageData();
   }, []);
 
-  React.useEffect(() => {
-    if (data) {
-      // At this point, the XHR request has set `data` to be an object.
-      // The user is definitely signed in or not signed in.
-      data.offlineSettings = OfflineSettingsData.read();
-      setSessionStorageData(data);
-
-      // Let's initialize the MDN Worker if applicable.
-      if (!window.mdnWorker && data?.offlineSettings?.offline) {
```

Comment:
And then, with an accurate `data.offlineSettings`, this block could update the `mdnWorker` to match? (Possibly outside of `user-context.tsx` - it doesn't feel very user contexty, it's dependent on the user context).